sdk/go/examples_test.go

```diff
@@ -140,7 +139,10 @@ func ExampleContainer_WithMountedCache() {
 	}
 	defer client.Close()
 
-	cacheKey := "example-" + time.Now().Format(time.RFC3339)
+	// Do not add a timestamp to the cacheKey because it
+	// would then be unique on each run and therefore
+	// not cache any results at all.
+	cacheKey := "example-cache"
```

Comment:
The goal with the timestamp is that the example runs the first iteration without a cache because of the unique key, and then the later 4 iterations use the same cache since they have the same key.

With this change, if you were to run this example multiple times the test should fail with output like `"0\n1\n2\n3\n4\n0\n1\n2\n3\n4\n"`.

I'm guessing this caused confusion though since normally you wouldn't have a time based cache key. Hmm. Maybe this could have a comment explaining that?

Reply:
I guess we could change the test so the `/cache/<epoch>.txt`  file is used instead?

The thing about comments is that people not always read them. It might be better to showcase the "regular" usage of `CacheVolume` and change the test code to avoid having weird outputs? 

WDYT Alex?

Reply:
> I guess we could change the test so the `/cache/<epoch>.txt` file is used instead?

Rather than `x.txt`? Makes sense to me :+1:

Reply:
OK, I'm trying to understand what you guys are saying, but am a bit confused because right now, as written, the tests pass:

```bash
$ go test -v -run ExampleContainer_WithMountedCache ./examples_test.go 
=== RUN   ExampleContainer_WithMountedCache
--- PASS: ExampleContainer_WithMountedCache (38.82s)
PASS
ok  	command-line-arguments	38.834s
```

Am I doing something wrong?

Reply:
The tests will always pass since they're not asserting anything, but what @vito is mentioning is that the `Printf` at the bottom of the function will print a funky output since the same cache and file is being reused every time.



> > I guess we could change the test so the `/cache/<epoch>.txt` file is used instead?
> 
> Rather than `x.txt`? Makes sense to me +1

great, I'll make the change Alex.

Reply:
I'm sorry, @marcosnils - I still don't understand... because as far as I can tell, that same string is currently being returned.

Anyway, to move this PR along, could you please click on "Add a suggestion" and type what you mean, and I will do it?

Reply:
> The tests will always pass since they're not asserting anything

I do not agree with this statement. If I change that line and run the tests again, look at what I get:

```bash
$ go test -v -run ExampleContainer_WithMountedCache ./examples_test.go 
=== RUN   ExampleContainer_WithMountedCache
--- FAIL: ExampleContainer_WithMountedCache (8.64s)
got:
"0\n1\n2\n3\n4\n"
want:
"bogus"
FAIL
FAIL	command-line-arguments	8.655s
FAIL
```

Reply:
So I would argue that the original test was misleading, because it passes with or without the timestamp.

Reply:
Can't suggest changes unfortunately since it's out of bounds, but it'd be to change line 155 to something like this:

```go
"echo $0 >> /cache/"+filename+".txt; cat /cache/"+filename+".txt",
```

where `filename` is something like `time.Now().Format(time.RFC3339)` assigned once outside of the loop so it's the same for each iteration.

That way we'll be showing a more canonical stable cache volume name, while still having consistent results for each run of the example.

> OK, I'm trying to understand what you guys are saying, but am a bit confused because right now, as written, the tests pass:

I would expect them to pass the first time, but fail if you were to run them again because each run would keep appending to the same cache.

Hope this helps, and thanks again!

Reply:
> I would expect them to pass the first time, but fail if you were to run them again because each run would keep appending to the same cache.

it's no that tests would fail since we're not asserting anything correct?

Reply:
ok, found out what's happening here. The reason why running the test twice doesn't make it fail is because the second time is using the cache and nothing gets really executed. 

Having said that, @vito's suggestion still applies here @gmlewis. Would you mind applying that single fix so we can merge? 

Thx!

Reply:
I've applied that suggestion (please correct me if I misunderstood), but now I can't get it to pass:

```
$ go test -v -run ExampleContainer_WithMountedCache ./examples_test.go 
=== RUN   ExampleContainer_WithMountedCache
--- FAIL: ExampleContainer_WithMountedCache (8.64s)
panic: input:1: container.from failed to do request: Head "https://registry-1.docker.io/v2/library/alpine/manifests/3.16.2": EOF

Please visit https://dagger.io/help#go for troubleshooting guidance. [recovered]
	panic: input:1: container.from failed to do request: Head "https://registry-1.docker.io/v2/library/alpine/manifests/3.16.2": EOF

Please visit https://dagger.io/help#go for troubleshooting guidance.

goroutine 1 [running]:
testing.(*InternalExample).processRunResult(0xc0000f9c18, {0x0, 0x0}, 0x0?, 0x0, {0x7aee80, 0xc000214080})
	/usr/local/go/src/testing/example.go:95 +0x64a
testing.runExample.func2()
	/usr/local/go/src/testing/run_example.go:59 +0x11c
panic({0x7aee80, 0xc000214080})
	/usr/local/go/src/runtime/panic.go:884 +0x213
command-line-arguments_test.ExampleContainer_WithMountedCache()
	/home/glenn/src/github.com/gmlewis/dagger/sdk/go/examples_test.go:160 +0x739
testing.runExample({{0x816955, 0x21}, 0x83f728, {0x80ede2, 0x12}, 0x0})
	/usr/local/go/src/testing/run_example.go:63 +0x2d0
testing.runExamples(0xc0000f9dd8, {0xad92a0?, 0x8, 0x0?})
	/usr/local/go/src/testing/example.go:44 +0x17d
testing.(*M).Run(0xc0000a2a00)
	/usr/local/go/src/testing/testing.go:1908 +0x6ef
main.main()
	_testmain.go:61 +0x1aa
FAIL	command-line-arguments	8.653s
FAIL
```

Reply:
seems unrelated. I think it's because "sometimes" the buildkit container networking gets borked and you have to remove the container. Mind giving that a try? `docker rm -fv $(docker ps --filter name=dagger-engine -q)`

Reply:
That worked!  Thank you, @marcosnils !

Do you want me to revert the changes to all the Python files?

Reply:
> Do you want me to revert the changes to all the Python files?

Yes, we should also update the examples in the python files as well the same way as we did in golang's.

Reply:
Whups, one moment, please... as there will be bad filename characters in the Python filenames... let me fix that.